boards: ti: lp_em_cc2340r5: Add support for debug and flash

gumulka opened the pull request:
Created a tool to generate the crc32 sum and write it inside the .hex file and put it as a post-build command step.

Now debugging and flashing the devboard using J-Link is possible.

gumulka committed:
boards: ti: lp_em_cc2340r5: Add support for flash & debug

Adding basic support for flashing and debugging. Not using the XDS110
debugger, as I have non on hand, but just a J-Link.

Signed-off-by: Fabian Pflug <fabian.pflug@gumulka.eu>

gumulka committed:
soc: ti: cc23x0: disable .bin generation

The .bin file is huge (>1GB) and will probably not be used by anyone if
there are much smaller alternatives. Because of that, disable it by
default for this soc. There is also no support for generating the
CRC32 checksum for the ccfg section in the binary.

Signed-off-by: Fabian Pflug <fabian.pflug@gumulka.eu>

gumulka commented on scripts/requirements-extras.txt:
```diff
@@ -9,6 +9,9 @@ gitpython>=3.1.41
 # helper for developers - check git commit messages
 gitlint
 
+# script to generate crc32 sums for TI CC23xx MCUs
+hex-modifier-tool
```
I use intelhex internally, but how exactly is this supposed to work with just intelhex?

gumulka replied:
Did you mean something like: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/silabs/silabs_siwx91x/siwx91x_isp_prepare.py?

Because that I can do and it would probably be better, yes.

nordicjm replied:
yes something like that, a dedicated script in the soc folder for generating the output

gumulka replied:
Made the script very simple and short for easy review :)

nordicjm commented on boards/ti/lp_em_cc2340r5/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+  # Insert crc32 checksums for ccfg into zephyr.hex image
+  COMMAND hex_modifier_tool -d ti_cc23xx ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+  # not done for .bin, as it >1GB and modifying it takes a long time.
+  )
```
that lpc file is wrong and should be fixed. Because the build system outputs files and expects those files to not be touched, if you take them and modify them, they need different names. You can see a ping pong behaviour by configuring a project then continually calling `ninja` over and over, the wrong thing that the lpc does means that each time it will regenerate the files over and over again rather than do nothing because the files are up to date

gumulka committed:
soc: ti: cc23x0: Add helper script to calculate crc32

As stated in [0] section 9.2 Customer Configuration:

	There are four different CRCs used to validate the CCFG data.
	One of the CRCs, the user record CRC, is optional and is the
	last four bytes of the 128B user record. The data over which
	the CRC is calculated starts at “Data Start Offset” from Table
	9-2 and ends at the "CRC Offset". CRC field width is 4 bytes.

Meaning the MCU will not start if the CRC fields are not correct.
TI's tools will automagically set these fields, but other tools have to
do it by hand. Therefore a new post-build-command is introduced to set
the fields at least in the .hex file by generating a new _crc32.hex
file.

[0] https://www.ti.com/lit/ug/swcu193a/swcu193a.pdf

Signed-off-by: Fabian Pflug <fabian.pflug@gumulka.eu>
